packages/react-components/react-carousel-preview/library/src/components/CarouselAutoplayButton/useCarouselAutoplayButton.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { slot, useControllableState, useIsomorphicLayoutEffect } from '@fluentui/react-utilities';
+import type { CarouselAutoplayButtonProps, CarouselAutoplayButtonState } from './CarouselAutoplayButton.types';
+import { useButton_unstable } from '@fluentui/react-button';
+import { PlayCircleRegular, PauseCircleRegular } from '@fluentui/react-icons';
+import { useEventCallback } from '@fluentui/react-utilities';
+import { mergeCallbacks } from '@fluentui/react-utilities';
+import { useCarouselContext_unstable as useCarouselContext } from '../CarouselContext';
+import { ARIAButtonElement } from '@fluentui/react-aria';
+
+/**
+ * Create the state required to render CarouselAutoplayButton.
+ *
+ * The returned state can be modified with hooks such as useCarouselAutoplayButtonStyles_unstable,
+ * before being passed to renderCarouselAutoplayButton_unstable.
+ *
+ * @param props - props from this instance of CarouselAutoplayButton
+ * @param ref - reference to root HTMLDivElement of CarouselAutoplayButton
+ */
+export const useCarouselAutoplayButton_unstable = (
+  props: CarouselAutoplayButtonProps,
+  ref: React.Ref<ARIAButtonElement>,
+): CarouselAutoplayButtonState => {
+  const [autoplay, setAutoplay] = useControllableState({
+    state: props.autoplay,
+    defaultState: props.defaultAutoplay,
```

Comment:
nit IMO `running` + `paused` like in Web Animations (https://developer.mozilla.org/en-US/docs/Web/API/Animation/playState#running) are more intuitive